src/backend/test.rs

```diff
@@ -343,6 +343,76 @@ impl Backend for TestBackend {
     fn flush(&mut self) -> io::Result<()> {
         Ok(())
     }
+
+    #[cfg(feature = "scrolling-regions")]
+    fn scroll_region_up(&mut self, region: std::ops::Range<u16>, scroll_by: u16) -> io::Result<()> {
```

Comment:
This method feels a bit more dense than it actually is because of the quantity of conditionals. Do you think it would help the readability / maintainability to flip the nesting to just a single layer and use early returns? e.g.:
```rust
        if region.start > 0 && cells_to_scroll_by >= cell_region.len() {
            // The scroll amount is large enough to clear the whole region.
            self.buffer.content[cell_region].fill_with(Default::default);
            return Ok(());
        }
```

Reply:
Okay, I rewrote it with early returns. Let me know what you think.

Reply:
I'm almost always a fan of early returns over nesting, so this is much better.